src/engine-actions/get-payment-channel-network-address.js

```diff
@@ -11,27 +11,24 @@ const HOST_DELIMITER = '@'
 /**
  * Returns the payment channel network address for this node
  *
- * @param {Object} [opts={}]
- * @param {boolean} [opts.includeHost=true]
  * @returns {string}
  */
-async function getPaymentChannelNetworkAddress ({ includeHost = true } = {}) {
-  const { uris = [] } = await getInfo({ client: this.client })
-  const uri = uris[0]
-
-  if (!uri) {
-    throw new Error(`LND has no uri avaialble`)
+async function getPaymentChannelNetworkAddress () {
+  const {
+    identityPubkey,
+    uris = []
+  } = await getInfo({ client: this.client })
+
+  // The uri will only exist if a user has set the `--external-ip` flag on the
+  // LND instance. If this uri does not exist, we will simply use the lnd public key
+  if (!uris.length) {
```

Comment:
Will `identityPubkey` always exist?

Reply:
@martineehrlich yeah it should always exist.

Reply:
@martineehrlich just to shield us from API changes, its probably worth it to check if it exists anyways